proofs/tree.go

```diff
@@ -298,7 +299,7 @@ func (doctree *DocumentTree) GetLeafByProperty(prop string) (int, *LeafNode) {
 func (doctree *DocumentTree) PropertyOrder() []string {
 	propOrder := []string{}
 	for _, leaf := range doctree.leaves {
-		propOrder = append(propOrder, leaf.Property)
+		propOrder = append(propOrder, leaf.Property.Name())
```

Comment:
This index should probably be duplicated for both types to allow lookup by human-readable values and bytes.

Reply:
Not quite. We use this index to create the proofs. I.e. given a property name, tell me which leaf in the tree a merkle proof should be created for. Given that we now support properties with human readable and compact version, we should have two indexes so that we can create a proof from either input. I think the right solution would be to create another method for the second index: 

`func (doctree *DocumentTree) CompactPropertyOrder() [][]byte`

Reply:
Actually []uint64 makes more sense.

Reply:
@lucasvo Could I see an example of where `PropertyOrder` is used like you said? I changed this last night to return `[]Property` and I want to see if that fits your use-case, or if a separate function is really required. I think `[]Property` is better because you should be able to give either the compact or human-readable name, and then get the leaf node from that, and then compute the hash using either method, not necessarily the method that was used to find the node in the first place. But let me know if I'm wrong about that or if that use case doesn't make any sense.

Reply:
This method `DocumentTree.CreateProof` should support both formats. It uses it internally. Have you added a test case for creating a proof with the binary format already?